features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/slowframes/UISlownessMetricDispatcher.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.rum.internal.metric.slowframes
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.InternalLogger.Target
+import com.datadog.android.rum.configuration.SlowFramesConfiguration
+import com.datadog.tools.annotation.NoOpImplementation
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicInteger
+
+@NoOpImplementation
+internal interface UISlownessMetricDispatcher {
+
+    fun onViewCreated(viewId: String)
+
+    fun incrementSlowFrameCount(viewId: String)
+
+    fun incrementIgnoredFrameCount(viewId: String)
+
+    fun incrementFreezeFrameCount(viewId: String)
+
+    fun sendMetric(viewId: String)
+}
+
+internal class DefaultUISlownessMetricDispatcher(
+    private val config: SlowFramesConfiguration,
+    private val internalLogger: InternalLogger,
+    private val samplingRate: Float = DEFAULT_SAMPLING_RATE
+) : UISlownessMetricDispatcher {
+
+    internal data class SlowFramesTelemetry(
+        var slowFramesCount: AtomicInteger = AtomicInteger(0),
+        var ignoredFramesCount: AtomicInteger = AtomicInteger(0),
+        var frozenFramesCount: AtomicInteger = AtomicInteger(0)
```

Comment:
it seems that these ones can be `val`?

```suggestion
        val slowFramesCount: AtomicInteger = AtomicInteger(0),
        val ignoredFramesCount: AtomicInteger = AtomicInteger(0),
        val frozenFramesCount: AtomicInteger = AtomicInteger(0)
```